src/nix/command.cc

```diff
@@ -61,15 +66,48 @@ void StorePathsCommand::run(ref<Store> store)
     }
 
     else {
+        if (!recursive && (includeBuildDeps || includeEvalDeps))
+            throw UsageError("--build and --eval require --recursive");
+
         for (auto & p : toStorePaths(store, realiseMode, installables))
             storePaths.push_back(p.clone());
 
         if (recursive) {
+            if (includeEvalDeps) {
+                for (auto & i : installables) {
+                    auto state = getEvalState();
+
+                    for (auto & b : i->toBuildables()) {
+                        if (!b.drvPath)
+                            throw UsageError("Cannot find eval references without a derivation path");
```

Comment:
Print what's `b`

Reply:
There's not a good way to print what b is without the drv path. We could show just "out" path, I guess:

https://github.com/NixOS/nix/blob/c9d0cf7e02d4b3af3c027d7d74d5caa3c8963d26/src/nix/installables.hh#L13-L14

Reply:
(but we don't have the store here so we can't know location nix store is at

Reply:
Printing at least the outputs would be desired otherwise there's no way to recover from this error.

Reply:
Added so the error message is now:

```
error: Cannot find build references for 'hello-2.10' without a derivation path
```